--- origsrc/squeak-3.10/platforms/unix/vm-display-X11/Makefile.in	2006-10-18 11:17:11.000000000 -0500
+++ src/squeak-3.10/platforms/unix/vm-display-X11/Makefile.in	2012-08-01 18:27:14.122640600 -0500
@@ -34,7 +34,7 @@
 TARGET		= vm-display-X11$a
 OBJS		= sqUnixX11$o sqUnixMozilla$o
 
-XCFLAGS		= $(X_CFLAGS)
+XCFLAGS		= $(X_CFLAGS) -DSUGAR
 
 XINCLUDES	= [includes] $(X_INCLUDES) \
 		  -I$(topdir)/platforms/Cross/plugins/FilePlugin \
