--- origsrc/squeak-3.10/platforms/unix/config/make.ext.in	2003-08-07 02:31:46.000000000 -0500
+++ src/squeak-3.10/platforms/unix/config/make.ext.in	2012-08-01 17:12:27.119998700 -0500
@@ -11,5 +11,6 @@ COMPILE		= $(LIBTOOL) --mode=compile \
 		  $(INCLUDES) $(XINCLUDES) $(CPPFLAGS) $(XCPPFLAGS) -c -o
 LINK		= $(LIBTOOL) --mode=link \
 		  $(CC) $(CFLAGS) $(XCFLAGS) \
-		  $(LDFLAGS) $(XLDFLAGS) -avoid-version -module -rpath $(plgdir) -o
+		  -Wl,$(top_builddir)/libsqueak.a \
+		  $(LDFLAGS) $(XLDFLAGS) -avoid-version -module -no-undefined -rpath $(plgdir) -o
 RANLIB		= :
--- origsrc/squeak-3.10/platforms/unix/config/make.prg.in	2011-03-28 11:23:11.000000000 -0500
+++ src/squeak-3.10/platforms/unix/config/make.prg.in	2012-08-01 17:10:25.813060300 -0500
@@ -11,5 +11,6 @@ COMPILE		= $(CC) $(CFLAGS) $(XCFLAGS) \
 		  $(LDFLAGS) $(XLDFLAGS) -export-dynamic -c -o
 LINK		= $(LIBTOOL) --mode=link \
 		  $(CC) $(CFLAGS) $(XCFLAGS) \
+		  -Wl,--out-implib,lib$@.a \
 		  $(LDFLAGS) $(XLDFLAGS) -export-dynamic -R$(libdir) -o
 RANLIB		= :
--- origsrc/squeak-3.10/platforms/unix/config/Makefile.in	2006-10-18 11:17:11.000000000 -0500
+++ src/squeak-3.10/platforms/unix/config/Makefile.in	2012-08-01 17:06:34.668839600 -0500
@@ -43,7 +43,7 @@ all : $(squeak) plugins squeak.1 $(npsqu
 # VM
 
 $(squeak) : config.sh $(SQLIBS) version.o
-	$(LINK) $(squeak) $(SQLIBS) version.o $(LIBS) [plibs] vm/vm.a
+	$(LINK) $(squeak) $(SQLIBS) version.o $(LIBS) [plibs]
 	@echo
 	@size $(squeak)
 	@echo
@@ -114,7 +114,7 @@ NPLIBS		= $(NPOBJS) -framework CoreAudio
 
 NPSqueak : config.sh $(SQLIBS) version.o $(NPOBJS)
 	gcc -bundle -flat_namespace -undefined suppress -o NPSqueak \
-	  $(NPLIBS) $(SQLIBS) version.o $(LIBS)  vm/vm.a
+	  $(NPLIBS) $(SQLIBS) version.o $(LIBS)
 
 # manual page
 
--- origsrc/squeak-3.10/platforms/unix/config/Makefile.install	2006-09-18 10:38:07.000000000 -0500
+++ src/squeak-3.10/platforms/unix/config/Makefile.install	2012-08-01 17:37:54.944385200 -0500
@@ -17,7 +17,7 @@ uninstall : uninstall-squeak uninstall-p
 
 install-squeak : $(squeak)
 	$(MKINSTALLDIRS) $(ROOT)$(plgdir)
-	$(INSTALL_PROG) $(squeak) $(ROOT)$(plgdir)
+	$(LIBTOOL) --mode=install $(INSTALL_PROG) $(squeak) $(ROOT)$(plgdir)
 	$(MKINSTALLDIRS) $(ROOT)$(bindir)
 	if test -n "$(VM_APP_ICONS)"; then \
 	  $(INSTALL_DATA) $(ICONS) $(ROOT)$(VM_APP_ICONS); \
--- origsrc/squeak-3.10/platforms/unix/plugins/ImmX11Plugin/Makefile.inc	1969-12-31 18:00:00.000000000 -0600
+++ src/squeak-3.10/platforms/unix/plugins/ImmX11Plugin/Makefile.inc	2012-08-01 18:19:49.957235800 -0500
@@ -0,0 +1,1 @@
+XLDFLAGS=  ../vm-display-X11/vm-display-X11.la
--- origsrc/Squeak-3.10-6/platforms/unix/plugins/SqueakFFIPrims/Makefile.in	2008-09-02 13:49:28.000000000 -0500
+++ src/Squeak-3.10-6/platforms/unix/plugins/SqueakFFIPrims/Makefile.in	2012-08-05 17:43:57.714726700 -0500
@@ -12,6 +12,7 @@ TARGET	= SqueakFFIPrims$a
 OBJS	= SqueakFFIPrims$o ffi-test$o $(FFI_O)
 
 XINCLUDES	= [includes]
+XLDFLAGS	= -lffi
 
 $(TARGET) : $(OBJS) Makefile
 	$(LINK) $(TARGET) $(OBJS)
--- origsrc/Squeak-3.10-6/platforms/unix/plugins/XDisplayControlPlugin/Makefile.inc	2008-09-02 13:49:31.000000000 -0500
+++ src/Squeak-3.10-6/platforms/unix/plugins/XDisplayControlPlugin/Makefile.inc	2012-08-05 17:22:16.834629100 -0500
@@ -1,3 +1,3 @@
 XCPPFLAGS= -I$(topdir)/platforms/Cross/plugins/FilePlugin -I$(topdir)/platforms/Cross/plugins/SocketPlugin
 XINCLUDES= $(X_INCLUDES)
-XLDFLAGS= $(X_LIBS)
+XLDFLAGS= ../vm-display-X11/vm-display-X11.la $(X_LIBS)
