--- origsrc/squeak-3.10/platforms/unix/config/configure	2011-03-28 11:55:56.000000000 -0500
+++ src/squeak-3.10/platforms/unix/config/configure	2012-08-01 17:59:34.503715800 -0500
@@ -29986,20 +29986,18 @@ plibs=""
 rm -f vm-sound-OSS.sub vm-sound-OSS.lib
 # -*- sh -*-
 
-{ echo "$as_me:$LINENO: checking for Advanced Linux Sound Architecture" >&5
-echo $ECHO_N "checking for Advanced Linux Sound Architecture... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking for Open Sound System Architecture" >&5
+echo $ECHO_N "checking for Open Sound System Architecture... $ECHO_C" >&6; }
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <alsa/asoundlib.h>
+#include <sys/soundcard.h>
 int
 main ()
 {
-(void)snd_pcm_open;
-  ;
   return 0;
 }
 _ACEOF
--- origsrc/squeak-3.10/platforms/unix/vm-sound-OSS/acinclude.m4	2006-09-18 09:35:52.000000000 -0500
+++ src/squeak-3.10/platforms/unix/vm-sound-OSS/acinclude.m4	2012-08-01 17:48:59.908419000 -0500
@@ -1,9 +1,3 @@
 # -*- sh -*-
 
-AC_MSG_CHECKING([for Advanced Linux Sound Architecture])
-AC_TRY_COMPILE([#include <alsa/asoundlib.h>],[(void)snd_pcm_open;],[
-  AC_MSG_RESULT(yes)
-],[
-  AC_MSG_RESULT(no)
-  AC_PLUGIN_DISABLE
-])
+AC_CHECK_HEADERS(sys/soundcard.h,,AC_PLUGIN_DISABLE)
--- origsrc/squeak-3.10/platforms/unix/vm-sound-OSS/Makefile.inc	2006-09-18 09:35:52.000000000 -0500
+++ src/squeak-3.10/platforms/unix/vm-sound-OSS/Makefile.inc	1969-12-31 18:00:00.000000000 -0600
@@ -1 +0,0 @@
-PLIBS=-lasound
