--- origsrc/squeak-3.10/platforms/unix/src/plugins/DBusPlugin/DBusPlugin.c	2008-07-26 13:48:52.000000000 -0500
+++ src/squeak-3.10/platforms/unix/src/plugins/DBusPlugin/DBusPlugin.c	2012-08-01 17:17:23.793967500 -0500
@@ -27,7 +27,6 @@ by VMMaker 3.8b6
 // was #undef EXPORT(returnType) but screws NorCroft cc
 #define EXPORT(returnType) static returnType
 #endif
-#include "aio.h"
 	#define DBUS_API_SUBJECT_TO_CHANGE
 #include <dbus/dbus.h>
 	typedef struct sqDBusData {
