--- origsrc/Squeak-3.10-6/platforms/unix/vm/debug.c	2012-08-05 18:10:05.043087800 -0500
+++ src/Squeak-3.10-6/platforms/unix/vm/debug.c	2012-08-05 16:57:58.076107300 -0500
@@ -1,9 +1,10 @@
-#include "debug.h"
 
 #include <stdio.h>
 #include <stdarg.h>
 #include <string.h>
 
+#include "debug.h"
+
 void __sq_dprintf(const char *fmt, ...)
 {
   va_list ap;
--- origsrc/Squeak-3.10-6/platforms/unix/vm-sound-NAS/sqUnixSoundNAS.c	2008-09-02 13:49:35.000000000 -0500
+++ src/Squeak-3.10-6/platforms/unix/vm-sound-NAS/sqUnixSoundNAS.c	2012-08-05 17:00:26.792497000 -0500
@@ -33,6 +33,8 @@
 #include <audio/audiolib.h>
 #include <assert.h>
 
+#include "debug.h"
+
 #ifdef DEBUG
 # define dprintf printf
 #else
